UsbDk/ControlDevice.cpp

```diff
@@ -1030,7 +1036,7 @@ void CUsbDkRedirection::NotifyRedirectionRemovalStarted()
 
 bool CUsbDkRedirection::WaitForDetachment()
 {
-    auto waitRes = m_RedirectionRemoved.Wait(true, -SecondsTo100Nanoseconds(120));
+    auto waitRes = m_RedirectionRemoved.Wait(true, -SecondsTo100Nanoseconds(1));
```

Comment:
You're changing timeout from 2 minutes to 1 second and this definitely solves 2-minutes stuck of the client. However the problem is that 1 second may be not enough for detachment.